sugoi-api-core/src/main/java/fr/insee/sugoi/core/realm/RealmProvider.java

```diff
@@ -41,10 +44,9 @@
    *
    * @param realmName, the realm name to return (case insensitive)
    * @return the realm found
-   * @throws RealmNotFoundException
    */
   @Cacheable(value = "Realm", key = "#realmName")
-  public Realm load(String realmName) throws RealmNotFoundException;
+  public Optional<Realm> load(String realmName) throws RealmNotFoundException;
```

Comment:
Do we cache an optional ? What if we add a realm along the way, will the optional be updated ?

Reply:
Good question i will test !!

Reply:
Damn, the optionnal is not updated :'(

Reply:
Erratum is not fully updated

Reply:
We can update the realm but not the userstorage in the realm